whistl/datasets.py

```diff
@@ -442,16 +431,84 @@ def __len__(self):
         return len(self.labels)
 
 
+def create_sample_to_platform_mapping(metadata):
+    '''Generate a dictionary mapping each sample to the platform that generated it
+
+    Arguments
+    ---------
+    metadata: json
+        A json object containing the metadata for a study
+
+    Returns
+    -------
+    sample_to_platform: dict
+        A dictionary mapping each sample accession to the platform used to sequence the sample
+    '''
+    sample_to_platform = {}
+
+    experiment_metadata = metadata['experiments']
+    sample_metadata = metadata['samples']
+
+    sample_list = []
+    for experiment in experiment_metadata:
+        try:
+            experiment_samples = experiment_metadata[experiment]['sample_accession_codes']
+            sample_list.extend(experiment_samples)
+        except KeyError as e:
+            # If an experiment doesn't have any samples for some reason, skip it
+            print(e)
+            pass
+
+
+    for sample in sample_list:
+        sample_to_platform[sample] = sample_metadata[sample]['refinebio_platform'].lower()
+
+    return sample_to_platform
+
+
+def subset_expression_by_platform(expression_df, platforms, sample_to_platform):
+    '''Subset a dataframe to contain only the samples produced by the given platform(s)
+
+    Arguments
+    ---------
+    expression_df: pandas Dataframe
+        A dataframe where the rows are genes and the columns are sample
+    platforms: list of str
+        A list of the platforms whose samples should be kept
+    sample_to_platform: dict
+        A dict mapping sample accessions to their corresponding platforms
+
+    Returns
+    -------
+    subset_df: pandas DataFrame
+        The subset of the dataframe passed in containing only samples from the given studies
+    '''
+    platform_set = set(platforms)
+    samples_to_keep = []
+
+    all_samples = expression_df.columns
+
+    for sample in all_samples:
+        if sample_to_platform[sample] in platform_set:
+            samples_to_keep.append(sample)
+
+    subset_df = expression_df.loc[:, samples_to_keep]
+
+    return subset_df
+
+
 class CompendiumDataset(Dataset):
     '''A dataset of one or more studies pulled from the refine.bio human compendium'''
 
-    def __init__(self, studies, classes, sample_to_label, metadata_path, compendium_path, encoder):
+    def __init__(self, groups, classes, sample_to_label, metadata_path, compendium_path, encoder,
```

Comment:
It might be my github but why is there a new line near the last argument? Also consider adding whitespace to header since the number of arguments have grown.

Reply:
The newline was to keep the function definition from exceeding the line length limit. I added more whitespace so it's no longer an issue